src/v/redpanda/admin/api-doc/debug_bundle.json

```diff
@@ -0,0 +1,256 @@
+        }
+      },
+      "422": {
+        "description": "Format error",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "500": {
+        "description": "Internal server error",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      }
+    }
+  },
+  "get": {
+    "description": "Returns the status of a debug bundle process.  If one was never kicked off and bundle does not exist, this will return 409",
+    "summary": "Returns the status of a debug bundle process.",
+    "operationId": "get_debug_bundle",
+    "produces": [
+      "application/json"
+    ],
+    "responses": {
+      "200": {
+        "description": "OK",
+        "schema": {
+          "$ref": "#/definitions/get_bundle_status"
+        }
+      },
+      "401": {
+        "description": "Unauthorized",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "403": {
+        "description": "Forbidden",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "409": {
+        "description": "Conflict",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      }
+    }
+  }
+},
+"/v1/debug/bundle/{jobid}": {
+  "delete": {
+    "description": "Attempts to kill a running bundle process.  Providing an incorrect `job_id` returns a 422.  If the process is not currently running, the request will return a 409 error.  Internal errors will return a 500 status.",
+    "summary": "Abort a running bundle process.",
+    "operationId": "delete_debug_bundle",
+    "consumes": [
+      "application/json"
+    ],
+    "parameters": [
+      {
+        "name": "jobid",
+        "in": "path",
+        "required": true,
+        "type": "string"
+      }
+    ],
+    "produces": [
+      "application/json"
+    ],
+    "responses": {
+      "204": {
+        "description": "No content"
+      },
+      "401": {
+        "description": "Unauthorized",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "403": {
+        "description": "Forbidden",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "409": {
+        "description": "Conflict",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "422": {
+        "description": "Format Error",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      },
+      "500": {
+        "description": "Internal error",
+        "schema": {
+          "$ref": "#/definitions/error_body"
+        }
+      }
+    }
+  }
+},
+"/v1/debug/bundle/file/{filename}": {
+  "get": {
+    "description": "Returns the generated debug bundle ZIP file",
+    "summary": "Returns the generated debug bundle ZIP file",
+    "operationId": "get_debug_bundle_file",
+    "parameters": [
+      {
+        "name": "filename",
```

Comment:
this API seems to allow fetching any file from any path?   Why expose the file system at all?  we can retrieve the bundle by the bundle ID.

Reply:
> this API seems to allow fetching any file from any path? Why expose the file system at all? we can retrieve the bundle by the bundle ID.

Yes, strong validation would be required at a a minimum here; to ensure that the supplied filename is known to be a debug bundle. That part of the backend has not been written yet.